tests/components/garmin_connect/test_config_flow.py

```diff
@@ -0,0 +1,122 @@
+"""Test the Garmin Connect config flow."""
+from unittest.mock import patch
+
+from garminconnect import (
+    GarminConnectAuthenticationError,
+    GarminConnectConnectionError,
+    GarminConnectTooManyRequestsError,
+)
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.garmin_connect import config_flow
+from homeassistant.components.garmin_connect.const import DOMAIN
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+MOCK_CONF = {
+    CONF_ID: "First Lastname",
+    CONF_USERNAME: "my@email.address",
+    CONF_PASSWORD: "mypassw0rd",
+}
+
+
+@pytest.fixture(name="mock_garmin_connect")
+def mock_garminconnect():
+    """Mock Garmin."""
+    with patch("homeassistant.components.garmin_connect.config_flow.Garmin") as garmin:
+        garmin().get_full_name.return_value = "First Lastname"
+        yield garmin
+
+
+def init_config_flow(hass):
+    """Init a configuration flow."""
+    flow = config_flow.GarminConnectConfigFlowHandler()
+    flow.hass = hass
+    return flow
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_user(user_input=None)
```

Comment:
Start the flow like this without user input:
https://github.com/home-assistant/home-assistant/blob/80a55360dc0cba19f2b7ccfdc84ea369ae368f73/tests/components/hue/test_config_flow.py#L140-L142

Reply:
This works, adapted.